code/controllers/subsystem/polling.dm

```diff
@@ -0,0 +1,216 @@
+SUBSYSTEM_DEF(polling)
+	name = "Polling"
+	flags = SS_BACKGROUND | SS_NO_INIT
+	wait = 1 SECONDS
+	runlevels = RUNLEVEL_GAME
+	/// List of polls currently ongoing, to be checked on next fire()
+	var/list/datum/candidate_poll/currently_polling
+	/// Whether there are active polls or not
+	var/polls_active = FALSE
+	/// Number of polls performed since the start
+	var/total_polls = 0
+	/// The poll that's closest to finishing
+	var/datum/candidate_poll/next_poll_to_finish
+
+/datum/controller/subsystem/polling/fire()
+	if(!polls_active)
+		return
+	if(!currently_polling) // if polls_active is TRUE then this shouldn't happen, but still..
+		currently_polling = list()
+
+	for(var/poll in currently_polling)
+		var/datum/candidate_poll/running_poll = poll
+		if(running_poll.time_left() <= 0)
+			polling_finished(running_poll)
+
+/datum/controller/subsystem/polling/proc/poll_candidates(question, role, check_jobban, poll_time = 30 SECONDS, ignore_category = null, flash_window = TRUE, list/group = null, pic_source, role_name_text)
+	if(group.len == 0)
+		return list()
+	if(role && !role_name_text)
+		role_name_text = role
+	if(role_name_text && !question)
+		question = "Do you want to play as [full_capitalize(role_name_text)]?"
+	if(!question)
+		question = "Do you want to play as a special role?"
+	log_game("Polling candidates [role_name_text ? "for [role_name_text]" : "\"[question]\""] for [poll_time / 10] seconds")
+
+	// Start firing
+	polls_active = TRUE
```

Comment:
No reason to have this extra parameter if we can derive it from `total_polls > 0`